src/main/java/io/cryostat/rules/RuleProcessor.java

```diff
@@ -255,13 +255,12 @@ private void startRuleRecording(ConnectionDescriptor connectionDescriptor, Rule
                     RecordingOptionsBuilder builder =
                             recordingOptionsBuilderFactory
                                     .create(connection.getService())
-                                    .name(rule.getRecordingName())
-                                    .toDisk(true);
+                                    .name(rule.getRecordingName());
                     if (rule.getMaxAgeSeconds() > 0) {
-                        builder = builder.maxAge(rule.getMaxAgeSeconds());
+                        builder = builder.maxAge(rule.getMaxAgeSeconds()).toDisk(true);
                     }
                     if (rule.getMaxSizeBytes() > 0) {
-                        builder = builder.maxSize(rule.getMaxSizeBytes());
+                        builder = builder.maxSize(rule.getMaxSizeBytes()).toDisk(true);
```

Comment:
Why are rules with `maxAge` or `maxSize`  always stored to disk?

Reply:
https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Recording.html#setMaxAge(java.time.Duration)

maxAge/maxSize control the size of the disk repository for JFR data that gets flushed to disk. If toDisk=false, maxAge/maxSize have no effect.

Reply:
The in-memory JFR ring buffers have a relatively small fixed size, so it's almost certain that a Cryostat user's use cases would include enabling toDisk. If they're configuring maxAge/maxSize it seems reasonable to assume they intend toDisk=true as well.